### PR TITLE
ADEN-11536

### PR DESCRIPTION
**Pull request by Tomasz-at-Fandom**

### CODE REVIEW

File: platforms/shared/sequential-messaging/domain/services/new-sequence-state-handler.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export class NewSequenceStateHandler implements SequenceStateHandlerInterface {
+	sequentialAdsIds: string[] = [];
+
+	constructor(sequentialMessagingConfig: SequentialMessagingConfig) {
+		for (const item of Object.keys(sequentialMessagingConfig)) {
+			this.sequentialAdsIds.push(item);
+		}
+	}
+
+	handleState(sequence: Sequence): boolean {
```
Comment (bckatarzyna) [inline platforms/shared/sequential-messaging/domain/services/new-sequence-state-handler.ts:14]: Is it ok that `handleState` can have different types across the handlers? And I'm not sure if this name goes with `boolean` type, detector seemed to be better in this case

File: spec/platforms/shared/sequential-messaging/domain/sequence-detector.spec.ts [hunk 2]
```diff
@@ -1,20 +1,23 @@
 import { expect } from 'chai';
-import { NewSequenceDetector } from '../../../../../platforms/shared/sequential-messaging/domain/services/new-sequence-detector';
+import { NewSequenceStateHandler } from '../../../../../platforms/shared/sequential-messaging/domain/services/new-sequence-state-handler';
 import { SequentialMessagingConfig } from '../../../../../platforms/shared/sequential-messaging/domain/data-structures/sequential-messaging-config';
+import { Sequence } from '../../../../../platforms/shared/sequential-messaging/domain/data-structures/sequence';
 
 describe('New Sequence Detector', () => {
```
Comment (bckatarzyna) [inline spec/platforms/shared/sequential-messaging/domain/sequence-detector.spec.ts:6]: `Detector`/`Handler`?

File: spec/platforms/shared/sequential-messaging/infrastructure/sequential-messaging-config-store.spec.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ describe('Sequence Messaging Config Store', () => {
 			1234567890: 'invalid value',
 		});
 		const smConfigStore_invalidLengthValue = makeInstantConfigServiceSpy();
-		smConfigStore_invalidLengthValue.get.returns({ 1234567890: { length: [4] } });
+		smConfigStore_invalidLengthValue.get.returns({ 1234567890: { lastStepId: [4] } });
```
Comment (bckatarzyna) [inline spec/platforms/shared/sequential-messaging/infrastructure/sequential-messaging-config-store.spec.ts:16]: Does `invalidLengthValue` still make sense?

File: spec/platforms/shared/sequential-messaging/domain/sequence-continuation-handler.spec.ts [hunk 4]
```diff
@@ -2,61 +2,63 @@ import { makeUserStateStoreSpy } from '../test_doubles/state-store.spy';
...
 import { expect } from 'chai';
-import { assert } from 'sinon';
+import sinon, { assert } from 'sinon';
 import { makeTargetingManagerSpy } from '../test_doubles/targeting-manager.spy';
+import { Sequence } from '../../../../../platforms/shared/sequential-messaging/domain/data-structures/sequence';
+import { SequenceEndStateHandler } from '../../../../../platforms/shared/sequential-messaging/domain/services/sequence-end-state-handler';
 
 const configStoreSample = {
 	5854346762: {
 		length: '4',
```
Comment (bckatarzyna) [inline spec/platforms/shared/sequential-messaging/domain/sequence-continuation-handler.spec.ts:12]: Is length still needed?

File: platforms/shared/sequential-messaging/domain/data-structures/sequential-messaging-config.ts [hunk 5]
```diff
@@ -1,3 +1,6 @@
 export interface SequentialMessagingConfig {
-	[sequentialAdId: string]: { length: number | string; targeting: Record<string, unknown> };
+	[sequentialAdId: string]: {
+		lastStepId: number | string;
```
[the remark this answers is not in the document]
Comment (Tomasz-at-Fandom) [reply platforms/shared/sequential-messaging/domain/data-structures/sequential-messaging-config.ts:3]: Yea I don't remember exactly now, I'll get rid of the number type here and see if it all works well.
I vaguely recall loosening this up due to some external input that I did not understand at first.

File: spec/platforms/shared/sequential-messaging/domain/sequence-continuation-handler.spec.ts (hunk 4 above)
Comment (Tomasz-at-Fandom) [reply spec/platforms/shared/sequential-messaging/domain/sequence-continuation-handler.spec.ts:12]: Nope, good catch!

File: spec/platforms/shared/sequential-messaging/domain/sequence-detector.spec.ts (hunk 2 above)
Comment (Tomasz-at-Fandom) [reply spec/platforms/shared/sequential-messaging/domain/sequence-detector.spec.ts:6]: Yea I'm gonna go back to the Detector name, I was planning a bigger change in that class but I scrapped it.

File: spec/platforms/shared/sequential-messaging/infrastructure/sequential-messaging-config-store.spec.ts (hunk 3 above)
Comment (Tomasz-at-Fandom) [reply spec/platforms/shared/sequential-messaging/infrastructure/sequential-messaging-config-store.spec.ts:16]: Oh yea good catch again! You should play baseball ⚾  :D

File: platforms/shared/sequential-messaging/domain/services/new-sequence-state-handler.ts (hunk 1 above)
Comment (Tomasz-at-Fandom) [reply platforms/shared/sequential-messaging/domain/services/new-sequence-state-handler.ts:14]: Yea I'm gonna go back to the Detector name, I was planning a bigger change in that class but I scrapped it.

**Review by bckatarzyna — APPROVED** [review]

**Pull request merged**